core/src/main/java/juzu/impl/common/Tools.java

```diff
@@ -194,7 +197,11 @@ public boolean hasNext() {
 
           @Override
           public T next() {
-            if (!hasNext()) {
+            if(next != null) {
```

Comment:
The hasNext method will avoid getting a null object in "next()" method. But I agree even that we shouldn't redefine the original iterator. But using the default iterator, it could return a null object, which is handled by this specific iterator.